### PR TITLE
Add select dropdown for CC provider selection page with CC iteration

### PR DESCRIPTION
**Pull request by LeKeve**
## Description

This PR adds a select dropdown to the CC provider selection when the CC iterations feature flag is on.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#28373

## Testing done
- local, unit testing

### CC provider selection with select dropdown
Screen Shot 2021-08-27 at 7 33 55 AM

### CC provider selection request location error
Screen Shot 2021-08-27 at 7 34 07 AM

### Design Assets
All spec copy is FPO. Use copy in copy doc.
- [Overview - dropdown](https://www.sketch.com/s/f97eb709-584d-4d37-a111-485828bcfafc/a/EL98PER#Inspector)
- [Dropdown sorting selector states](https://www.sketch.com/s/f97eb709-584d-4d37-a111-485828bcfafc/a/QbrLaqY)
- [Mobile spec](https://www.sketch.com/s/f97eb709-584d-4d37-a111-485828bcfafc/a/Kvwjd8g#Inspector)
- [Tablet spec](https://www.sketch.com/s/f97eb709-584d-4d37-a111-485828bcfafc/v/akdEVZ/a/wLMWGYq/r/ovlW73#Inspector)
- [Desktop spec](https://www.sketch.com/s/f97eb709-584d-4d37-a111-485828bcfafc/v/akdEVZ/a/qe5Wzy2/r/ovlW73#Inspector)

```
Feature: VAOS ProviderSortVariant on <CommunityCareProviderSelectionPage>

    Scenario: should display list of providers when choose a provider clicked
      Given the CC iteration flag is on
      When the user clicks the choose a provider button
      Then providers should be displayed

    Scenario: should notify user that the browser is blocked from using current location information
      Given the CC iteration flag is on
        And the user denies geolocation
      When the user selects to sort providers by distance from current location
      Then an error location alert should be displayed

    Scenario: should sort provider addresses by distance from current location in ascending order when current location is selected
      Given the CC iteration flag is on
      When the user selects to sort providers by distance from current location
      Then providers should be displayed in ascending order by distance from current location

    Scenario: should allow user to retry fetching location when it is blocked
      Given the CC iteration flag is on
        And the user denies geolocation
        And the error location alert is displayed
      When the user attempts to search by current location again
      Then providers should be displayed by distance from current location

    Scenario: should sort providers by distance from selected facility in ascending order
      Given the CC iteration flag is on
      When the user selects to sort providers by distance from a specific facility
      Then providers should be displayed in ascending order by distance from the chosen facility

    Scenario: should default to first ccEnabledSystem if user is missing a residential address
      Given the CC iteration flag is on
        And the user does not have a residential address
      When the user tries to choose a provider
      Then the select options should default to sort by distance from the first CC enabled facility
```

## Acceptance criteria
Given user is in CC flow
And there is no provider selected
When the user opens the provider preference page
Then the provider question content matches the copy doc
And the provider question design matches the design spec and templates

Given a user is in the CC flow
And the user has a home address on file
And the user is on the provider preference page
When user selects Choose a provider
Then the list sort options are home address, current location, and facility city/state
And the default sort is home address
And the copy and design match the copy and design docs

Given a user is in the CC flow
And the user is on the provider preference page
And the provider list is expanded
When the user changes the sort method
Then order of providers in list is updated to the new sort method

Given a user is in the CC flow
And the user is on the provider preference page
And the provider list is expanded
And the user changed the sort method to current location
When the browser fails to get the current location
Then show location blocked and retry message

Given a user is in the CC flow
And the user is on the provider preference page
And the user does not have a home address on file
When the user selects Choose a provider
Then the list sort options are current location and facility city/state
And the default sort is the first facility city/state

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx [hunk 1]
```diff
@@ -39,7 +45,11 @@ export default function ProviderSelectionField({
   useEffect(
     () => {
       if (mounted && Object.keys(formData).length === 0) {
-        scrollAndFocus('.va-button-link');
+        if (showCCIterations) {
```
Comment (LeKeve) [inline src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx:48]: @jbalboni The original implementation focuses on the residential/current location sort button, so I set the focus to the select in the cc iteration. If you think it makes more sense to focus on the first provider in the list I can update. I think it makes sense either way.

Comment (jbalboni) [reply src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx:48]: I went back and looked at the what it's doing currently in master. It looks like it's 

- Focus on Choose provider h2 after you click "+ Choose a provider"
- When you click on "use your address/use current location", then focus jumps to loading spinner, then back to h2 when complete

I think doing both of those with the dropdown flow is what we want.

I don't see any intentional focusing on the sort buttons, though it does look like we're inadvertently doing that in this branch when the toggle is off (but not on master).

File: src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSortVariant.jsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  const {
+    address,
+    ccEnabledSystems,
+    communityCareProviderList,
+    currentLocation,
+    requestLocationStatus,
+    selectedCCFacility,
+    sortMethod,
+  } = useSelector(selectProviderSelectionInfo, shallowEqual);
+  const [selectedSortMethod, setSelectedSortMethod] = useState(sortMethod);
```
[the remark this answers is not in the document]
Comment (jbalboni) [reply src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSortVariant.jsx:28]: Ok. We tend to run into bugs when we trying to sync redux and local state, so I'd like to try to get rid of it, though not necessarily in this ticket, if this is working for now. We may need to put a spinner somewhere to wait until the parent location data is fetched.

File: src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx (hunk 1 above)
Comment (LeKeve) [reply src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx]: It looks like the inadvertent focusing on the sort/dropdown buttons is due to my refactoring of the `CommunityCareProviderSelectionPage` to use `useSelector`. When I revert that change the focus works as intended, it looks like there is an extra render from updating the passed `formData` in the schemaform.

I fixed the focusing by removing a redundant useEffect in the `ProviderSelectionField` component.

File: src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx [hunk 3]
```diff
@@ -33,16 +33,7 @@ export default function ProviderSelectionField({
         scrollAndFocus('#providerPostSelectionHeader');
       }
     },
-    [showProvidersList],
-  );
-
-  useEffect(
```
Comment (LeKeve) [inline src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx:39]: This `useEffect` was used to focus on the `Choose a provider` button after selecting/removing a provider. Adding `providerSelected` to the `useEffect` above achieves the same effect and removes the extra focusing.

**Comment by jbalboni** [comment]
Focus handling looks fixed and things overall appear to be working, just noticed that the distance isn't showing up correctly for parent sites:

Screen Shot 2021-08-31 at 9 51 21 AM

**Comment by LeKeve** [comment]
@jbalboni Fixed distance display bug for sort by facility distance.

**Review by jbalboni — APPROVED** [review]

**Pull request merged**